rfcs/0000-kuadrant-release-process.md

```diff
@@ -0,0 +1,207 @@
+Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.
+
+A more detailed explanation of the versioning scheme can be found in the [Semantic Versioning](https://semver.org/) website.
+
+By releasing a new version of Kuadrant, we mean releasing a new version of every component in the suite, with their
+corresponding semantic versioning, and being the version of the **Kuadrant Operator** the one that defines the version of
+the whole suite.
+
+The technical details of how to release each component are out of the scope of this RFC and could be found in the
+[Kuadrant components CI/CD](https://github.com/Kuadrant/architecture/pull/41) RFC.
+
+## Handover to QA
+
+Probably the most important and currently missing step in the release process is the handover to the Quality Assurance
+(QA) team. The QA team is responsible for testing the different components of the Kuadrant suite, and they need to
+be aware of the new version of the suite that is going to be released, what are the changes that are included, bug fixes
+and new features in order they can plan their testing processes accordingly. The handover to QA should happen once the
+release candidate is ready, and it's been tested by the Engineering team. The QA team should be notified in advance
+of the release, and they should be able to test the release candidate before the actual release happens.
+
+There is an ideal time to hand over to the QA team for testing, especially since we are using GitHub for orchestration,
+we could briefly define it in the following steps:
+
+1. Complete Development Work: The engineering team completes their work included in the milestone and it's merged to main branch.
+2. Prepare Release: The engineering team creates a release branch from the main branch, including the particular changes
+that configure the release (manifests, version bump, etc.)
+3. Tag Release: The engineering team tags the current state of the main branch as a release candidate. This creates a
+snapshot of the codebase that can be tested without affecting the main branch and builds the deliverables.
+4. Notify QA Team: At this time, the engineering team can notify the QA team that the deliverables for the release candidate
+are ready for testing. This notification can be done through comments on GitHub issues or via other communication channels
+like Slack.
+5. Testing: The QA team tests the release candidate, checking for any bugs or issues. They can use GitHub's issue tracking
+system to report any problems they find.
+6. Iterate: Based on the feedback from the QA team, the development team makes any necessary adjustments and repeats the
+process until the release candidate is deemed ready for production.
+7. Publish Release: Once the release candidate is ready, the engineering team publishes the release. This means that the
+Github release draft is published, and the artifacts are uploaded to the corresponding registries.
+
+## Cadence
```

Comment:
This question might not belong here but looking through the links I was not able to find the answer or the correct place for the question. 

If we are releasing the Kuadrant Operator every two weeks, what is our support policy? How many version are we going to back port security and bug fixes to?

Reply:
It's a valid question, and important one to have it registered in the RFC. I'd say that we should include not only engineering and QA, but what would be the expectation from Product as well.

Reply:
I will add it to the "Unresolved questions" section